src/main/java/org/opentestsystem/rdw/reporting/service/impl/ItemServiceImpl.java

```
@@ -0,0 +1,96 @@
+package org.opentestsystem.rdw.reporting.service.impl;
+
+import org.opentestsystem.rdw.reporting.model.Item;
+import org.opentestsystem.rdw.reporting.model.Rubric;
+import org.opentestsystem.rdw.reporting.repository.ArtifactRepository;
+import org.opentestsystem.rdw.reporting.service.ItemService;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.xpath.*;
+import java.io.*;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ItemServiceImpl implements ItemService {
+    private ArtifactRepository artifactRepository;
+
+    public ItemServiceImpl(ArtifactRepository artifactRepository) {
+        this.artifactRepository = artifactRepository;
+    }
+
+    @Override
+    public Item getById(String id) {
+        try {
+            Document xmlDoc = GetXmlDocument(id);
+
+            XPath xPath = XPathFactory
+                .newInstance()
+                .newXPath();
+
+            Node rubriclistNode = (Node) xPath
+                .compile("/itemrelease/item/content/rubriclist")
+                .evaluate(xmlDoc, XPathConstants.NODE);
+
+            NodeList nodes = (NodeList) xPath.evaluate("./rubric", rubriclistNode, XPathConstants.NODESET);
+
+            Item item = new Item()
+                .builder()
+                .id(id)
+                .rubrics(GetRubrics(xPath, nodes)                )
+                .build();
+
+            return item;
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (SAXException e) {
+            e.printStackTrace();
+        } catch (XPathExpressionException e) {
+            e.printStackTrace();
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
+        }
```

services generally will wrap native exceptions with a service-specific exception and then declare that exception in a throws clause. And then the logging will eventually be done centrally in a catch all exception handler or method advice. Actually, in enterprise Java it is frowned upon to use print stack trace at all. I can show you an example of all this.